### ISSUE TITLE
sae_manifold_fit and ManifoldSAE.fit() silently ignore top_k / target_k

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#241]
## Repro

```python
import numpy as np
import gamfit

rng = np.random.default_rng(0)
X = rng.standard_normal((20, 3))
fit = gamfit.sae_manifold_fit(
    Z=X,
    n_atoms=3,
    atom_basis="periodic",
    atom_dim=1,
    assignment="softmax",
    top_k=1,
    max_iter=5,
    random_state=0,
)
A = np.asarray(fit.assignments)
print(((A > 1e-12).sum(axis=1)).max())  # >>> 3, expected 1
```

And the torch wrapper:

```python
import torch, gamfit.torch as gt

cfg = gt.ManifoldSAEConfig(
    input_dim=6,
    n_atoms=4,
    intrinsic_rank=1,
    atom_manifold="circle",
    atom_basis="fourier",
    sparsity={"kind": "softmax_topk", "target_k": 1},
)
sae = gt.ManifoldSAE(cfg).double()
fit = sae.fit(torch.randn(20, 6, dtype=torch.float64), max_iter=3)
print(((np.asarray(fit.assignments) > 1e-12).sum(axis=1)).max())  # >>> 4
```

## Root cause

`gamfit/_binding.py:79-89` accepts `top_k` then explicitly discards it:

```python
def sae_manifold_fit_minimal(z, k_atoms, atom_basis, atom_dim, ..., top_k, ...):
    ...
    del top_k  # reserved for future top-K activation; not yet plumbed
    return underlying(z, atom_basis, atom_dim, ...)  # top_k not forwarded
```

The underlying Rust `sae_manifold_fit_auto` (`crates/gam-pyffi/src/lib.rs:9988`) has no `top_k` parameter — there is nowhere to forward it to even if `_binding.py` tried.

`gamfit/torch/manifold_sae.py:683-695` `ManifoldSAE.fit(...)` reads `cfg.sparsity` but does NOT pass `cfg.sparsity.target_k` into `_closed_form_sae_manifold_fit`. Torch `forward()` (line 510) applies a top-k gate at runtime, but `.fit()`'s closed-form path leaves all atoms active. The two paths disagree on the same `cfg.sparsity` contract.

## Proposed fix

Pick one:

1. Implement top-k in the Rust closed-form assignment path (`sae_manifold_fit_auto`), accept `top_k` in the pyffi signature, forward it from `_binding.py` and from `ManifoldSAE.fit()` (via `cfg.sparsity.target_k`).
2. Reject loudly at the wrapper boundary: `_binding.sae_manifold_fit_minimal` should raise `NotImplementedError` for any `top_k != 0/None`, and `ManifoldSAE.fit()` should refuse `sparsity.kind == "softmax_topk"`.

Silent drop is the worst of both — option (2) is the minimum-acceptable interim contract.

## Notes

RED tests added in `tests/test_sae_manifold_top_k_issue.py`.

**Comment by SauersML** [comment]
Fixed in fe6949cb4. `gamfit.sae_manifold_fit` now applies a hard top-k mask post-Rust (`_apply_top_k_mask`); `gamfit.torch.ManifoldSAE.fit()` forwards `cfg.sparsity.target_k` to the closed-form path as `top_k`. All 4 RED tests in `tests/test_sae_manifold_top_k_issue.py` pass.
